src/Microsoft.DotNet.VersionTools/BuildManifest/SupplementaryUploadRequest.cs

```diff
@@ -6,8 +6,20 @@ namespace Microsoft.DotNet.VersionTools.BuildManifest
 {
     public class SupplementaryUploadRequest
     {
+        /// <summary>
+        /// Path, relative to the primary upload dir or absolute with a leading '/'.
+        /// </summary>
         public string Path { get; set; }
 
         public string Contents { get; set; }
+
+        public string GetAbsolutePath(string currentPath)
```

Comment:
What's the difference between this and `Path.Combine`?

Reply:
Hmm, I thought `Path.Combine` would give me back a platform-dependent path, but apparently not. I guess the difference then is that this method allows `GetInvalidPathChars`, but I don't need those. I'll see about switching over.

Reply:
Gah, for that comment I tried it out with `("ignored/base", "/absolute/dir")` and saw that it didn't *convert* any of my slashes, but forgot the primary case is `("build-info/foo", "bar/baz")` which does insert a `\` between the subpaths on Windows. Keeping this method as-is.